core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -561,36 +666,62 @@ public static Object sum( ValueMetaInterface metaA, Object dataA, ValueMetaInter
     return plus( metaA, dataA, metaB, dataB );
   }
 
+  /**
+   * @deprecated Use {@link ValueDataUtil#loadFileContentInBinary(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
   public static Object loadFileContentInBinary( ValueMetaInterface metaA, Object dataA ) throws KettleValueException {
+    Object content = null;
+    try {
+      content = loadFileContentInBinary( metaA, dataA, true );
+    } catch ( KettleFileNotFoundException e ) {
+      throw new KettleValueException();
+    }
+    return content;
+  }
+
+  /**
+   *
+   * @param metaA
+   *   The ValueMetaInterface
+   * @param dataA
+   *   Filename
+   * @param failIfNoFile
+   *   Indicates if the transformation should fail if no file is found
+   * @return File's content in binary
+   * @throws KettleValueException
+   * @throws KettleFileNotFoundException
+   */
+  public static Object loadFileContentInBinary( ValueMetaInterface metaA, Object dataA, boolean failIfNoFile )
+          throws KettleValueException, KettleFileNotFoundException {
     if ( dataA == null ) {
+      if ( failIfNoFile ) {
```

Comment:
This actually is not a file not found.... it's a file path/name not provided.... I'm not sure the file not found exception suits this case